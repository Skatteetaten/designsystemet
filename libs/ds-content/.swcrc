{
  "jsc": {
    "target": "es2021",
    "parser": {
      "syntax": "typescript",
      "tsx": true,
      "decorators": true,
      "dynamicImport": true
    },
    "transform": {
      "react": {
        "runtime": "automatic",
        "importSource": "react"
      },
      "decoratorMetadata": true,
      "legacyDecorator": true
    },
    "keepClassNames": true,
    "externalHelpers": true,
    "loose": true
  },
  "module": {
    "type": "es6"
  },
  "sourceMaps": true,
  "exclude": ["jest.config.ts",".*\\.spec.tsx?$",".*\\.test.tsx?$","./src/jest-setup.ts$","./**/jest-setup.ts$",".*.js$"]
}
