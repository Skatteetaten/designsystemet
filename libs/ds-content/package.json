{
  "name": "@skatteetaten/ds-content",
  "groupId": "no.skatteetaten.aurora",
  "version": "1.4.0",
  "license": "Apache-2.0",
  "homepage": "https://www.skatteetaten.no/stilogtone/designsystemet/",
  "type": "module",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/Skatteetaten/designsystemet.git"
  },
  "publishConfig": {
    "registry": "https://nexus.sits.no/repository/npm-internal/"
  },
  "peerDependencies": {
    "react": "^18",
    "@skatteetaten/ds-core-utils": "1.4.0",
    "@skatteetaten/ds-typography": "1.4.0",
    "@skatteetaten/ds-icons": "1.4.0",
    "react-i18next": "^14",
    "i18next": "^23"
  },
  "scripts": {},
  "dependencies": {},
  "devDependencies": {}
}
