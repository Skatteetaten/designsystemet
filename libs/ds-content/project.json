{
  "name": "ds-content",
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/ds-content/src",
  "projectType": "library",
  "tags": [],
  "implicitDependencies": ["!ds-dev-config"],
  "targets": {
    "lint": {
      "executor": "@nx/eslint:lint",
      "outputs": ["{options.outputFile}"]
    },
    "build": {
      "executor": "@nx/rollup:rollup",
      "outputs": ["{options.outputPath}"],
      "options": {
        "outputPath": "dist/libs/ds-content",
        "tsConfig": "libs/ds-content/tsconfig.lib.json",
        "project": "libs/ds-content/package.json",
        "entryFile": "libs/ds-content/src/index.ts",
        "external": ["react/jsx-runtime"],
        "rollupConfig": ["libs/ds-content/rollup.config.js"],
        "compiler": "swc",
        "assets": [
          {
            "glob": "libs/ds-content/README.md",
            "input": ".",
            "output": "."
          },
          {
            "glob": "LICENSE",
            "input": ".",
            "output": "."
          }
        ],
        "babelUpwardRootMode": true
      }
    },
    "test": {
      "executor": "@nx/jest:jest",
      "outputs": ["{workspaceRoot}/coverage/{projectRoot}"],
      "options": {
        "jestConfig": "libs/ds-content/jest.config.ts"
      }
    },
    "stylelint": {
      "executor": "nx-stylelint:lint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": [
          "libs/ds-content/**/*.css",
          "libs/ds-content/**/*.scss"
        ]
      }
    },
    "tsc": {
      "executor": "@skatteetaten/ds-dev-config:tsCheck",
      "options": {
        "tsConfig": ["tsconfig.lib.json", "tsconfig.spec.json"]
      },
      "dependsOn": [
        {
          "projects": "ds-core-designtokens",
          "target": "build"
        }
      ]
    },
    "set-version": {
      "executor": "nx:run-commands",
      "options": {
        "command": "npm version {args.packageVersion} --no-git-tag",
        "cwd": "libs/ds-content"
      }
    },
    "publish": {
      "executor": "nx:run-commands",
      "options": {
        "command": "npm publish dist/libs/ds-content {args.switches}"
      }
    }
  }
}
