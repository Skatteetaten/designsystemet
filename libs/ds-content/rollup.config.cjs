const path = require('path');

const createRollupConfig = require('../ds-dev-config/build/rollup-config-factory.cjs');

const srcDir = path.resolve(__dirname, `src`);
const pathToTsConfig = 'libs/ds-content/tsconfig.lib.json';
const pathToCSS = path.resolve(__dirname, `**/*.{css,scss}`);
const outputDir = path.resolve('./', `dist/libs/ds-content/`);

module.exports = (config) =>
  createRollupConfig(config, srcDir, pathToTsConfig, pathToCSS, outputDir);
