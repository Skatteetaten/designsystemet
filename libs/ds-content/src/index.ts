export * from './Card/Card';
export * from './Card/Card.types';
export * from './CardAlert/CardAlert.types';
export * from './DescriptionList/DescriptionList';
export * from './DescriptionList/DescriptionList.types';
export * from './Panel/Panel';
export * from './Panel/Panel.types';
