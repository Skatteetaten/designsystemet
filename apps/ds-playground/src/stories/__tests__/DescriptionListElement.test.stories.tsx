import { JSX } from 'react';

import { Meta, StoryObj } from '@storybook/react';
import { expect, within } from '@storybook/test';

import { DescriptionList } from '@skatteetaten/ds-content';

const meta = {
  component: DescriptionList.Element,
  title: 'Tester/DescriptionList/Element',
  argTypes: {
    // Baseprops
    key: { table: { disable: true } },
    ref: { table: { disable: true } },
    className: {
      table: { disable: true },
    },
    id: { table: { disable: true } },
    lang: { table: { disable: true } },
    'data-testid': { table: { disable: true } },
    // Props
    children: {
      table: { disable: true },
    },
    term: {
      table: { disable: true },
    },
  },
  args: {
    term: 'Ansvarlig fagperson',
    children: 'Kenneth Performance',
  },
  decorators: [
    (Story): JSX.Element => (
      <dl>
        <Story />
      </dl>
    ),
  ],
} satisfies Meta<typeof DescriptionList.Element>;
export default meta;
type Story = StoryObj<typeof meta>;

const defaultArgs = {};

export const WithRef = {
  name: 'With Ref (FA1)',
  args: {
    ...defaultArgs,
    'data-testid': '123ID',
    ref: (instance: HTMLDListElement | null): void => {
      if (instance) {
        instance.id = 'dummyIdForwardedFromRef';
      }
    },
  },
  argTypes: {
    ref: { table: { disable: false } },
  },
  parameters: { imageSnapshot: { disable: true } },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const container = canvas.getByTestId('123ID');
    await expect(container).toHaveAttribute('id', 'dummyIdForwardedFromRef');
  },
} satisfies Story;

export const WithAttributes = {
  name: 'With Attributes (FA2-5)',
  args: {
    ...defaultArgs,
    id: 'htmlId',
    className: 'dummyClassname',
    lang: 'nb',
    'data-testid': '123ID',
  },
  argTypes: {
    id: { table: { disable: false } },
    className: { table: { disable: false } },
    lang: { table: { disable: false } },
    'data-testid': { table: { disable: false } },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const container = canvas.getByTestId('123ID');
    await expect(container).toHaveClass('dummyClassname');
    await expect(container).toHaveAttribute('id', 'htmlId');
    await expect(container).toHaveAttribute('lang', 'nb');
  },
} satisfies Story;
