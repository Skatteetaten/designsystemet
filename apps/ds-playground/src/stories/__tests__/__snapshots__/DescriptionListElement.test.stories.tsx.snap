// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tester/DescriptionList/Element WithAttributes play-test 1`] = `
<div data-test-block="true">
  <dl>
    <dt id="htmlId"
        class="_term_1bsgs_1 dummyClassname"
        lang="nb"
        data-testid="123ID"
    >
      Ansvarlig fagperson
    </dt>
    <dd class="_description_1bsgs_6">
      Kenneth Performance
    </dd>
  </dl>
</div>
`;

exports[`Tester/DescriptionList/Element WithRef play-test 1`] = `
<div data-test-block="true">
  <dl>
    <dt class="_term_1bsgs_1"
        data-testid="123ID"
        id="dummyIdForwardedFromRef"
    >
      Ansvarlig fagperson
    </dt>
    <dd class="_description_1bsgs_6">
      Kenneth Performance
    </dd>
  </dl>
</div>
`;
